cli/src/stake.rs

```diff
@@ -1624,6 +1633,7 @@ pub fn process_stake_authorize(
     } else {
         Message::new(&ixs, Some(&fee_payer.pubkey()))
     };
+    simulate_and_update_compute_unit_limit(rpc_client, &mut message)?;
```

Comment:
We don't want to make a network request if it's `sign_only` right?

Reply:
Since `compute_unit_limit` is set to `ComputeUnitLimit::Default`, the instruction creator won't tack on a `set_compute_unit_limit` instruction, and `simulate_and_update_compute_unit_limit` will no-op if there's no compute unit limit instruction.

Maybe we need to make that clearer though. We can rename `simulate_and_update_compute_unit_limit` to `simulate_and_update_compute_unit_limit_if_present` -- what do you think? Or we can wrap all calls to `simulate_and_update_compute_unit_limit` in a `if !sign_only {` block

Reply:
Renaming the function is fine with me but I would like the function to be a bit stricter about when it simulates then. It should only simulate when compute unit limit is set to some marker value I think.. otherwise a statically set compute unit limit would still trigger a simulation. Currently we use `MAX_COMPUTE_UNIT_LIMIT` but we could set it to `0` and replace it inside the function before simulating?

I think wrapping the calls could be nice and explicit too for people reading the code. We could check if `compute_unit_limit == ComputeUnitLimit::Simulated` before calling?

Reply:
I was having a tough time figuring out how to do this as clearly as possible, so I updated `simulate_and_update_compute_unit_limit` to take the `ComputeUnitLimit` enum, and added return case for calling the function when the limit isn't set to `Simulated`, which is unused. Let me know what you think!

Reply:
I think it's much cleaner, thanks!